[BUG] Unexpected loadManifest calls for new manifest files introduced in Next.js 16.2.0-canary

### Describe the bug

When building and previewing a Next.js app with next@canary (16.2.0-canary.53+) using @opennextjs/cloudflare, all dynamic
  routes return 500 Internal Server Error at runtime. The error is thrown by the CF Workers runtime:

  Error: Unexpected loadManifest(/.next/server/subresource-integrity-manifest.json) call!
  Error: Unexpected loadManifest(/.next/server/app/api/[route]/react-loadable-manifest.json) call!

  These manifest files were introduced in Next.js 16.2.0-canary and are not yet handled by @opennextjs/cloudflare, causing
  the worker to crash on any request that triggers loadManifests().

  **Environment**

  - @opennextjs/cloudflare: 1.16.5
  - next: 16.2.0-canary.53
  - wrangler: 4.67.0

### Steps to reproduce

 1. Create a Next.js app router project:
  pnpm create next-app my-app --app
  cd my-app
  2. Add @opennextjs/cloudflare and upgrade Next.js to canary:
  pnpm add @opennextjs/cloudflare
  pnpm add next@canary
  3. Create an open-next.config.ts and wrangler.jsonc as per the OpenNext Cloudflare docs.
  4. Build and preview:
  pnpm opennextjs-cloudflare build
  pnpm opennextjs-cloudflare preview
  5. Navigate to any dynamic/SSR route — you'll receive a 500 with the following in wrangler logs:
  ✘ [ERROR] ⨯ Error: Unexpected loadManifest(/.next/server/subresource-integrity-manifest.json) call!

### Expected behavior

Dynamic routes should render successfully. The new manifest files introduced in the canary
  (subresource-integrity-manifest.json, per-route react-loadable-manifest.json) should either be bundled into the worker or
  gracefully handled when not present.

### @opennextjs/cloudflare version

1.16.5

### Wrangler version

4.67.0

### next info output

```shell
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 24.5.0: Tue Apr 22 19:54:49 PDT 2025; root:xnu-11417.121.6~2/RELEASE_ARM64_T6000
  Available memory (MB): 32768
  Available CPU cores: 10
Binaries:
  Node: 20.19.2
  npm: 10.8.2
  Yarn: 1.22.22
  pnpm: 9.15.9
Relevant Packages:
  next: 16.2.0-canary.58 // Latest available version is detected (16.2.0-canary.58).
  eslint-config-next: N/A
  react: 19.1.0
  react-dom: 19.1.0
  typescript: 5.9.3
Next.js Config:
  output: N/A
```

### Additional context

_No response_

--- a/.changeset/fix-load-manifest-graceful-fallback.md
+++ b/.changeset/fix-load-manifest-graceful-fallback.md
@@ -10,12 +10,17 @@ generated, so the patched function threw at runtime, crashing dynamic routes wit
 
 Instead of a blanket catch-all, handle only the specific optional manifests from Next.js
 `route-module.ts`:
-- `react-loadable-manifest.json` (Turbopack per-route, not all routes have dynamic imports)
-- `subresource-integrity-manifest.json` (only when `experimental.sri` configured)
-- `server-reference-manifest.json` (App Router only)
-- `dynamic-css-manifest.json` (Pages Router + Webpack only)
-- `fallback-build-manifest.json` (only for `/_error` page)
-- `_client-reference-manifest.js` (optional for static metadata routes)
+
+- `react-loadable-manifest` (Turbopack per-route, not all routes have dynamic imports)
+- `subresource-integrity-manifest` (only when `experimental.sri` configured)
+- `server-reference-manifest` (App Router only)
+- `dynamic-css-manifest` (Pages Router + Webpack only)
+- `fallback-build-manifest` (only for `/_error` page)
+- `prefetch-hints` (new in Next.js 16.2)
+- `_client-reference-manifest.js` (optional for static metadata routes, evalManifest)
+
+Manifest matching strips `.json` before comparison since some Next.js constants omit
+the extension (`SUBRESOURCE_INTEGRITY_MANIFEST`, `DYNAMIC_CSS_MANIFEST`, etc.).
 
 Unknown manifests still throw to surface genuine errors.
 

--- a/packages/cloudflare/src/cli/build/patches/plugins/load-manifest.ts
+++ b/packages/cloudflare/src/cli/build/patches/plugins/load-manifest.ts
@@ -69,12 +69,19 @@ function loadManifest($PATH, $$$ARGS) {
   // Known optional manifests \u2014 Next.js loads these with handleMissing: true
   // (see vercel/next.js packages/next/src/server/route-modules/route-module.ts).
   // Return {} to match Next.js behaviour instead of crashing the worker.
-  if ($PATH.endsWith("react-loadable-manifest.json") ||
-      $PATH.endsWith("subresource-integrity-manifest.json") ||
-      $PATH.endsWith("server-reference-manifest.json") ||
-      $PATH.endsWith("dynamic-css-manifest.json") ||
-      $PATH.endsWith("fallback-build-manifest.json")) {
-    return {};
+  // Note: Some manifest constants in Next.js omit the .json extension
+  // (e.g. SUBRESOURCE_INTEGRITY_MANIFEST, DYNAMIC_CSS_MANIFEST), so we
+  // strip .json before matching to handle both forms.
+  {
+    const p = $PATH.replace(/\\.json$/, "");
+    if (p.endsWith("react-loadable-manifest") ||
+        p.endsWith("subresource-integrity-manifest") ||
+        p.endsWith("server-reference-manifest") ||
+        p.endsWith("dynamic-css-manifest") ||
+        p.endsWith("fallback-build-manifest") ||
+        p.endsWith("prefetch-hints")) {
+      return {};
+    }
   }
   throw new Error(\`Unexpected loadManifest(\${$PATH}) call!\`);
 }`,

